internal/engine/local/controller.go

```diff
@@ -152,6 +163,41 @@ func (c *Controller) start(ctx context.Context, spec ServeSpec, st store.RStore)
 
 	spanID := SpanIDForServeLog(proc.procNum)
 	proc.doneCh = c.execer.Start(ctx, spec.ServeCmd, logger.Get(ctx).Writer(logger.InfoLvl), statusCh, spanID)
+
+	if spec.ReadinessProbe != nil {
+		statusChangeFunc := processReadinessProbeUpdate(ctx, st, spec.ManifestName, proc.stillHasSameProcNum())
+		probeWorker, err := probeWorkerFromSpec(
+			c.probeManager,
+			spec.ReadinessProbe,
+			probe.WorkerOnStatusChange(statusChangeFunc))
+		if err != nil {
+			logger.Get(ctx).Warnf("Invalid readiness probe: %v", err)
+		} else {
+			proc.probeWorker = probeWorker
+			go proc.probeWorker.Run(ctx)
+		}
+	}
+}
+
+func processReadinessProbeUpdate(
+	ctx context.Context,
+	st store.RStore,
+	manifestName model.ManifestName,
+	stillHasSameProcNum func() bool) probe.StatusChangedFunc {
+	return func(status prober.Result, output string) {
+		if !stillHasSameProcNum() {
+			return
+		}
+		if status != prober.Success && output != "" {
+			w := logger.Get(ctx).Writer(logger.DebugLvl)
+			_, _ = io.WriteString(w, output)
```

Comment:
How does this actually look when it's logged? From a brief inspection, it looks like if an exec probe fails, we'll just log the command's output, without any text indicating it was a probe or that it failed. Or maybe my brief inspection was too brief. Either way, this seems like a good thing to verify/document with an integration test.

Reply:
No, you're absolutely right - I had this same thought. There's also a pretty good chance that it'll be a multiline string from the probe, e.g. imagine some failing logs from an `Exec` probe or HTML/unminified JSON from `HTTPGet` probe, so I think it might even make sense to prepend each _line_ in `output` with `[readiness-probe]` or something?

Reply:
Yep, makes sense. Fine with a TODO for now if you wanna come back with another PR for integration tests.

Reply:
Threw a TODO here in 2390d0eaeb4a46afb156b33d539c7b9e434fae81. I've actually already got the code to do what I suggested above but will PR it after as it's purely cosmetic and can be covered by the forthcoming integration tests.